src/Umbraco.Web.UI.Client/src/common/directives/validation/nodirtycheck.directive.js

```diff
@@ -10,13 +10,12 @@ function noDirtyCheck() {
         require: 'ngModel',
         link: function (scope, elm, attrs, ctrl) {
 
-            var alwaysFalse = {
-                    get: function () { return false; },
-                    set: function () { }
-                };
-            Object.defineProperty(ctrl, '$pristine', alwaysFalse);
-            Object.defineProperty(ctrl, '$dirty', alwaysFalse);
+            // If no attribute value is "false", then skip and use default behaviour.
+            var dirtyCheck = scope.$eval(attrs.noDirtyCheck) === false;
```

Comment:
There must be a native JS function to evaluate a boolean instead of depending on AngularJS to do it for us?

Reply:
Yes, probably. However I have noticed the `$eval` function is used several other places to parse attribute values.

Reply:
With the above changes in mind, shouldn't this be `var dirtyCheck = Object.toBoolean(attrs.noDirtyCheck) === false;` ?